src/components/tl/ucp/tl_ucp_coll.c

```diff
@@ -51,6 +53,123 @@ static ucc_status_t ucc_tl_ucp_coll_finalize(ucc_coll_task_t *coll_task)
     return UCC_OK;
 }
 
+static ucc_status_t ucc_tl_ucp_triggered_coll_complete(ucc_coll_task_t *parent_task,
+                                                       ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+
+    tl_info(task->team->super.super.context->lib,
+        "triggered collective complete. task:%p", coll_task);
+    return ucc_mc_ee_task_end(coll_task->ee_task, coll_task->ee->ee_type);
+}
+
+static ucc_status_t ucc_tl_ucp_event_trigger_complete(ucc_coll_task_t *parent_task,
+                                                      ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+
+    tl_info(task->team->super.super.context->lib, "event triggered. task:%p", coll_task);
+
+    coll_task->ee_task = parent_task->ee_task;
+    coll_task->post(coll_task);
+    if (coll_task->super.status == UCC_OK) {
+        return ucc_tl_ucp_triggered_coll_complete(coll_task, coll_task);
+    } else {
+
+        ucc_assert(coll_task->super.status = UCC_INPROGRESS);
+
+        if (coll_task->ee_task) {
+            ucc_event_manager_init(&coll_task->em);
+            coll_task->handlers[UCC_EVENT_COMPLETED] = ucc_tl_ucp_triggered_coll_complete;
+            ucc_event_manager_subscribe(&coll_task->em, UCC_EVENT_COMPLETED, coll_task);
+        }
+    }
+
+    return UCC_OK;
+}
+
+//TODO can we move this logic to CORE
+static ucc_status_t ucc_tl_ucp_ee_wait_for_event_trigger(ucc_coll_task_t *coll_task)
+{
+    ucc_ev_t *post_event;
+    ucc_status_t status;
+    ucc_ev_t *ev;
+    ucc_tl_ucp_task_t *task = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+
+    if (task->super.ev == NULL) {
+        if (task->super.ee->ee_type == UCC_EE_CUDA_STREAM) {
+            /* implicit event triggered */
+            task->super.ev = (ucc_ev_t *) 0xFFFF; /* dummy event */
+            task->super.ee_task = NULL;
+        } else if (UCC_OK == ucc_ee_get_event_internal(task->super.ee, &ev,
+                                                &task->super.ee->event_in_queue)) {
+            tl_info(task->team->super.super.context->lib,
+                    "triggered event arrivied. task:%p", coll_task);
+            task->super.ev = ev;
+            task->super.ee_task = NULL;
+        } else {
+            return UCC_OK;
+        }
+    }
+
+    if (task->super.ee_task == NULL) {
+        status = ucc_mc_ee_task_post(task->super.ee->ee_context,
+                                     task->super.ee->ee_type, &task->super.ee_task);
+        if (status != UCC_OK) {
+            tl_error(task->team->super.super.context->lib, "error in ee task post");
```

Comment:
also need to set task status
task->super.super.status = status

Reply:
fixed